qiskit_ionq/ionq_job.py

```diff
@@ -413,36 +415,46 @@ def _format_result(self, data):
         qiskit_header = decompress_metadata_string_to_dict(
             metadata.get("qiskit_header", None)
         )
-
         shots = int(metadata.get("shots") if metadata.get("shots").isdigit() else 1024)
-        job_result = {
-            "data": {},
-            "shots": shots,
-            "header": qiskit_header or {},
-            "success": success,
-        }
-        if self._status == jobstatus.JobStatus.DONE:
-            (counts, probabilities) = _build_counts(
-                data,
-                self._num_qubits,
-                self._clbits,
-                shots,
-                use_sampler=is_ideal_simulator,
-                sampler_seed=sampler_seed,
-            )
-            job_result["data"] = {
-                "counts": counts,
-                "probabilities": probabilities,
-                # Qiskit/experiments relies on this being present in this location in the
-                # ExperimentData class.
-                "metadata": qiskit_header or {},
+        job_result = [
+            {
+                "data": {},
+                "shots": shots,
+                "header": qiskit_header or {},
+                "success": success,
             }
+            for _ in range(self._num_circuits or 1)
+        ]
+        if self._status == jobstatus.JobStatus.DONE:
+            # to handle ionq returning different data structures for single and multiple circuits
+            if self._num_circuits > 1:
+                data = list(data.values())
+            else:
+                data = [data]
+            for idx in range(self._num_circuits):
+                (counts, probabilities) = _build_counts(
+                    data[idx],
+                    self._num_qubits,
+                    self._clbits,
```

Comment:
I think this will need to come from metadata as well,

Currently we set this into a top level "registers" field on the Job but this may vary from circuit to circuit.